Zend/zend_lazy_objects.h

```diff
@@ -0,0 +1,108 @@
+/*
+   +----------------------------------------------------------------------+
+   | Zend Engine                                                          |
+   +----------------------------------------------------------------------+
+   | Copyright (c) Zend Technologies Ltd. (http://www.zend.com)           |
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 2.00 of the Zend license,     |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available through the world-wide-web at the following url:           |
+   | http://www.zend.com/license/2_00.txt.                                |
+   | If you did not receive a copy of the Zend license and are unable to  |
+   | obtain it through the world-wide-web, please send a note to          |
+   | license@zend.com so we can mail you a copy immediately.              |
+   +----------------------------------------------------------------------+
+   | Authors: Arnaud Le Blanc <arnaud.lb@gmail.com>                       |
+   +----------------------------------------------------------------------+
+*/
+
+#ifndef ZEND_LAZY_OBJECT_H
+#define ZEND_LAZY_OBJECT_H
+
+#include "Zend/zend_types.h"
+
+/* Lazy object is a lazy proxy object */
+#define ZEND_LAZY_OBJECT_STRATEGY_PROXY                     (1<<0)
+
+/* Lazy object is a lazy ghost object */
+#define ZEND_LAZY_OBJECT_STRATEGY_GHOST                     (1<<1)
+
+/* Lazy object is initialized (info.u is an instance) */
+#define ZEND_LAZY_OBJECT_INITIALIZED                        (1<<2)
```

Comment:
This flag seems a bit unnecessary, since we're already storing this information on the object itself. It looks like this is only used in one place (apart from asserts), namely `zend_lazy_object_info_dtor_func()`, where it seems like it could just be replaced by a `info->flags & ZEND_LAZY_OBJECT_STRATEGY_PROXY` check.